server/src/main/java/org/elasticsearch/search/retriever/CompoundRetrieverBuilder.java

```diff
@@ -254,6 +254,7 @@ protected SearchSourceBuilder createSearchSourceBuilder(PointInTimeBuilder pit,
         }
         sortBuilders.add(new FieldSortBuilder(FieldSortBuilder.SHARD_DOC_FIELD_NAME));
         sourceBuilder.sort(sortBuilders);
+        sourceBuilder.skipInnerHits(true);
```

Comment:
Should we rely on the fact that the compound retriever will always set this? I mean why do we need to set this on the knn and standard retriever too? 
We should also ensure that implementations of the `CompoundRetrieverBuilder` calls `super.createSourceBuilder` first if they want to modify the source builder. `TextSimilarityRankRetrieverBuilder` does it own thing and already diverged (missing the shard id sort and now this new skip inner hits).

Reply:
That's a fair point! Will update to ensure that the parent `createSourceBuilder` will always be called, and remove the checks from the individual retrievers.

Reply:
Updated.